index.js

```diff
@@ -24,71 +24,99 @@ module.exports = (opts = {}, ctx) => ({
       .action(async (dir = '.') => {
         dir = join(process.cwd(), dir)
         try {
-          const nCtx = await dev({
+          const devContext = await dev({
             sourceDir: dir,
             clearScreen: false,
-            theme: opts.theme || '@vuepress/default'
+            theme: options.theme || '@vuepress/default'
           })
+
           logger.setOptions({ logLevel: 3 })
           logger.info(`Start to generate current site to PDF ...`)
+
           try {
-            await generatePDF(nCtx, {
-              port: nCtx.devProcess.port,
-              host: nCtx.devProcess.host,
-              sorter: opts.sorter
+            await generatePDF(devContext, {
+              port: devContext.devProcess.port,
+              host: devContext.devProcess.host,
+              base: devContext.base,
+              options
             })
           } catch (error) {
-            console.error(red(error))
+            logger.error(red(error))
           }
-          nCtx.devProcess.server.close()
+
+          devContext.devProcess.server.close()
           process.exit(0)
-        } catch (e) {
-          throw e
+        } catch (error) {
+          throw error
         }
       })
   }
 })
 
-async function generatePDF(ctx, {
+const applyFilter = (filter) => {
+  if (typeof filter === 'function') {
+    return (page) => filter(page.location, page)
+  }
+
+  return (page) => {
+    return !filter || filter.test(page.location)
+  }
+}
+
+const createOutputFilename = (dest, pluginConfig, fallback) => {
+  if (typeof dest === 'function') {
+    return dest(pluginConfig)
+  }
+
+  if (typeof dest === 'string') {
+    return dest
+  }
+
+  return `${pluginConfig.title || String(fallback)}.pdf`
+}
+
+async function generatePDF(context, {
   port,
```

Comment:
Not related to your changes but it would be great if you could include in your fork. on Windows this needs to be
```
 host: devContext.devProcess.displayHost,
```
See https://github.com/vuejs/vuepress/commit/4d5c50e4455fdea18fbd62225889f70654d4b0c0